src/dxtbx/nexus/__init__.py

```diff
@@ -404,9 +404,14 @@ def equipment_component_key(dependency):
                 origin -= nxdetector.beam_center_y.magnitude * pixel_size[1] * slow_axis
 
         # dxtbx requires image size in the order fast, slow - which is the reverse of what
-        # is stored in module.data_size
-        image_size = cast(Tuple[int, int], tuple(map(int, module.data_size[::-1])))
-        assert len(image_size) == 2
+        # is stored in module.data_size. Additionally, data_size can have more than 2
+        # dimensions, for multi-module detectors. So take the last two dimensions and reverse
+        # them.  Examples:
+        # [1,2,3,4][-1:-3:-1] --> [4, 3]
+        # [1,2,3][-1:-3:-1] --> [3, 2]
+        # [1,2][-1:-3:-1] --> [2, 1]
+        image_size = cast(Tuple[int, int], tuple(map(int, module.data_size[-1:-3:-1])))
+        assert len(image_size) == 2, image_size
```

Comment:
```suggestion
        # [1,2,3,4] --> [4, 3]
        # [1,2,3]   --> [3, 2]
        # [1,2]     --> [2, 1]
        image_size = (int(module.data_size[-1]), int(module.data_size[-2])
```
I think this is more explicit, and doesn't involve reading ranged negative slices (it perhaps should have been this before anyway)